static/app/views/alerts/rules/metric/ruleConditionsForm.tsx

```diff
@@ -618,76 +726,103 @@ class RuleConditionsForm extends PureComponent<Props, State> {
+                            ),
+                          }}
+                          onSearch={query => {
+                            onFilterSearch(query, true);
+                            onChange(query, {});
+                          }}
+                          onBlur={(query, {parsedQuery}) => {
+                            onFilterSearch(query, parsedQuery);
+                            onBlur(query);
+                          }}
+                        />
+                      ) : (
+                        <StyledSearchBar
+                          disallowWildcard={dataset === Dataset.SESSIONS}
+                          disallowFreeText={[
+                            Dataset.GENERIC_METRICS,
+                            Dataset.TRANSACTIONS,
+                          ].includes(dataset)}
+                          invalidMessages={{
+                            [InvalidReason.WILDCARD_NOT_ALLOWED]: t(
+                              'The wildcard operator is not supported here.'
+                            ),
+                            [InvalidReason.FREE_TEXT_NOT_ALLOWED]: t(
+                              'Free text search is not allowed. If you want to partially match transaction names, use glob patterns like "transaction:*transaction-name*"'
+                            ),
+                          }}
+                          customInvalidTagMessage={item => {
+                            if (dataset !== Dataset.GENERIC_METRICS) {
+                              return null;
+                            }
+                            return (
+                              <SearchInvalidTag
+                                message={tct(
+                                  "The field [field] isn't supported for performance alerts.",
+                                  {
+                                    field: <code>{item.desc}</code>,
+                                  }
+                                )}
+                                docLink="https://docs.sentry.io/product/alerts/create-alerts/metric-alert-config/#tags--properties"
+                              />
+                            );
+                          }}
+                          searchSource="alert_builder"
+                          defaultQuery={initialData?.query ?? ''}
+                          {...getSupportedAndOmittedTags(dataset, organization)}
+                          includeSessionTagsValues={dataset === Dataset.SESSIONS}
+                          disabled={disabled || isErrorMigration}
+                          useFormWrapper={false}
+                          organization={organization}
+                          placeholder={this.searchPlaceholder}
+                          onChange={onChange}
+                          query={initialData.query}
+                          // We only need strict validation for Transaction queries, everything else is fine
+                          highlightUnsupportedTags={
```

Comment:
Looks like you probably want to pass this same thing to `disallowUnsupportedFilters` in SearchQueryBuilder?